engine/src/main/java/org/terasology/rendering/world/WorldRendererImpl.java

```diff
@@ -207,8 +212,24 @@ private void initMaterials() {
     private void initRenderGraph() {
         // FIXME: init pipeline without specifying them as a field in this class
         NodeFactory nodeFactory = new NodeFactory(context);
+
+        // ShadowMap generation
+        FBOConfig shadowMapConfig =
+                new FBOConfig(ShadowMapNode.SHADOW_MAP, FBO.Type.NO_COLOR, shadowMapResolutionDependentFBOs).useDepthBuffer();
+        BufferClearingNode.RequiredData shadowMapClearingData =
+                new BufferClearingNode.RequiredData(shadowMapConfig, GL_DEPTH_BUFFER_BIT);
+        Node shadowMapClearingNode = nodeFactory.createInstance(BufferClearingNode.class, shadowMapClearingData);
```

Comment:
As mentioned in a new, earlier comment I'm just starting with the refactoring of the nodes to make them fully take advantage of the new architecture. So, I don't know just yet how many node types will be reusable to the point of requiring initial data to customize each instance. 

A number of node types will be instanced only once. But some others will be certainly instanced more than once.  For example right now the "blur" shader is applied five times (as far as I remember), in different circumstances (DoF, Bloom). Those will most likely become instances of the same BlurNode class, each instance initialized with slightly different parameters.

I can imagine every Node type requiring initial data will probably require some kind of FBO to work on. Perhaps the BaseInitialData class could just bring that information, other nodes extending it with their own additional data. 

I'm really thinking this should be tackled when I have more nodes "under the belt", so that I can establish a usage pattern. But I'd be happy to eventually have some kind of base class or an interface to constrain the parameter type.

All that been said, in the meantime I'll extract the inner class.